h2o-py/tests/pyunit_utils/utils_for_glm_tests.py

```diff
@@ -0,0 +1,150 @@
+
+    base_iteration = find_glm_iterations(grid_models[0])
+    num_models = len(grid_models)
+    best_model_ind = 0
+    model_indices = []
+    model_equality_constraints_values = []
+    model_lessthan_constraints_values = []
+    iterations = []
+    for ind in range(1, num_models):
+        curr_model = grid_models[ind]
+        curr_metric = grid_models[ind].model_performance()._metric_json[metric]
+        metric_diff = abs(base_metric-curr_metric)/abs(base_metric)
+        if metric_diff < epsilon:
+            curr_constraint_table = curr_model._model_json["output"]["linear_constraints_table"]
+            [equality_constraints_values, lessthan_constraints_values] = grab_constraint_values(
+                curr_constraint_table, cond_index, num_constraints)
+            # conditions used to choose the best model
+            if (sum(equality_constraints_values) < sum(best_equality_constraints)) and (sum(lessthan_constraints_values) < sum(best_lessthan_constraints)):
+                best_model_ind = ind
+                base_iteration = find_glm_iterations(curr_model)
+                best_equality_constraints = equality_constraints_values
+                best_lessthan_constraints = lessthan_constraints_values
+            model_equality_constraints_values.append(equality_constraints_values)
+            model_lessthan_constraints_values.append(lessthan_constraints_values)
+            model_indices.append(ind)
+            iterations.append(find_glm_iterations(curr_model))
+    print("Maximum iterations: {0} and it is from model index: {1}".format(base_iteration, best_model_ind))
+    print_model_hyperparameters(grid_models[best_model_ind], hyper_parameters)
+    return grid_models[best_model_ind]
+
+def print_model_hyperparameters(model, hyper_parameters):
+    print("***** Hyper parameter values for best model chosen are:")
+    params = model.actual_params
+    param_keys = hyper_parameters.keys()
+    actual_keys = params.keys()
+    for param in actual_keys:
+        if param in param_keys:
+            print("{0} value {1}.".format(param, params[param]))
+            
+        
+def grab_constraint_values(curr_constraint_table, cond_index, num_constraints):
+    equality_constraints_values = []
+    lessthan_constraints_values = []
+    for ind2 in range(0, num_constraints): # collect all equality constraint info
+        if curr_constraint_table.cell_values[ind2][cond_index]=="== 0":
+            equality_constraints_values.append(curr_constraint_table.cell_values[ind2][cond_index-1])
+        else:
+            if curr_constraint_table.cell_values[ind2][cond_index-1] < 0:
+                lessthan_constraints_values.append(0)
+            else:
+                lessthan_constraints_values.append(curr_constraint_table.cell_values[ind2][cond_index-1])
+    return [equality_constraints_values, lessthan_constraints_values]
+    
+
+def compare_tuple(original_tuple, new_tuple):
```

Comment:
I think `compare_tuple` name can be misleading here. Maybe `is_always_lower_than` or some similar name would be more appropriate.

Personally, I prefer to use generators/list comprehension and `all` or `any` to do similar things as I think it's easier to read than for loop. The function would then be on just two lines of code:

```python
assert len(original_tuple) == len(new_tuple)
return all(abs(orig) > abs(new) for orig, new in zip(original_tuple, new_tuple))
```

But that's just a matter of personal opinion so feel free to disregard that.

Reply:
Done.  Copied your idea.  Thanks.